snapcraft/cli/_options.py

```diff
@@ -65,49 +65,112 @@ def prompt_for_value(self, ctx):
         is_flag=True,
         help="Shells into the environment after the step has run.",
     ),
+]
+
+_SUPPORTED_PROVIDERS = ["host", "lxd", "multipass"]
+_HIDDEN_PROVIDERS = ["managed-host"]
+_PROVIDER_OPTIONS = [
     dict(
         param_decls="--destructive-mode",
         is_flag=True,
-        help="Forces snapcraft to try and use the current host to build.",
+        help="Forces snapcraft to try and use the current host to build (implies `--provider=host`).",
     ),
     dict(
         param_decls="--use-lxd",
         is_flag=True,
-        help="Forces snapcraft to use LXD to build.",
+        help="Forces snapcraft to use LXD to build (implies `--provider=lxd`).",
+    ),
+    dict(
+        param_decls="--provider",
+        envvar="SNAPCRAFT_BUILD_ENVIRONMENT",
```

Comment:
If https://github.com/snapcore/snapcraft/pull/2678/files#diff-1b5470a9f2e177fc1606479628098324R123 is done anyways, it probably shouldn't be here... I'd rather keep this envvar as internal, we should probably log.warning when being used (not this PR), but not expose it here either

Reply:
https://click.palletsprojects.com/en/7.x/api/
`show_envvar – controls if an environment variable should be shown on the help page. Normally, environment variables are not shown.`

Reply:
let's use this key